### ISSUE TITLE
Inconsistent backend responses

### ISSUE THREAD

**Issue by kelsos** [raiden-network/explorer#138]
@czepluch found that a token was not being displayed on the explorer page for kovan (kovan.explorer.raiden.network).

After some slight investigation it seems that the backend returns a different number of results on each request, and then 404s for a few and then doing the same all over again.

Without knowing how things work exactly, I would guess that the backend crashes and then tries to rebuild state until it crashes gain.

**Comment by palango** [comment]
Yes, the error is the following:
```
backend-kovan_1          | 12-08 16:34:27 - metrics_backend.metrics_service - CRITICAL - Unhandled exception. Terminating the program...
backend-kovan_1          | Traceback (most recent call last):
backend-kovan_1          |   File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
backend-kovan_1          |   File "/app/metrics_backend/utils/blockchain_listener.py", line 160, in _run
backend-kovan_1          |     self._update()
backend-kovan_1          |   File "/app/metrics_backend/utils/blockchain_listener.py", line 217, in _update
backend-kovan_1          |     self.filter_events(filters_confirmed, self.confirmed_callbacks)
backend-kovan_1          |   File "/app/metrics_backend/utils/blockchain_listener.py", line 284, in filter_events
backend-kovan_1          |     callback(decoded_event)
backend-kovan_1          |   File "/app/metrics_backend/metrics_service.py", line 216, in handle_token_network_created
backend-kovan_1          |     event_block_number,
backend-kovan_1          |   File "/app/metrics_backend/metrics_service.py", line 230, in create_token_network_for_address
backend-kovan_1          |     token_infos = get_token_info(token_contract)
backend-kovan_1          |   File "/app/metrics_backend/utils/token.py", line 31, in get_token_info
backend-kovan_1          |     get_token_name(token_contract),
backend-kovan_1          |   File "/app/metrics_backend/utils/token.py", line 9, in get_token_name
backend-kovan_1          |     return token_contract.functions.name().call()
backend-kovan_1          |   File "/usr/local/lib/python3.6/site-packages/web3/contract.py", line 1120, in call
backend-kovan_1          |     **self.kwargs
backend-kovan_1          |   File "/usr/local/lib/python3.6/site-packages/web3/contract.py", line 1369, in call_contract_function
backend-kovan_1          |     return_data = web3.eth.call(call_transaction, block_identifier=block_id)
backend-kovan_1          |   File "/usr/local/lib/python3.6/site-packages/eth_utils/functional.py", line 20, in inner
backend-kovan_1          |     return callback(fn(*args, **kwargs))
backend-kovan_1          |   File "/usr/local/lib/python3.6/site-packages/web3/eth.py", line 294, in call
backend-kovan_1          |     [transaction, block_identifier],
backend-kovan_1          |   File "/usr/local/lib/python3.6/site-packages/web3/manager.py", line 112, in request_blocking
backend-kovan_1          |     raise ValueError(response["error"])
backend-kovan_1          | ValueError: {'code': -32015, 'message': 'VM execution error.', 'data': 'Reverted 0x'}
```

**Comment by czepluch** [comment]
Yet another kudos to @arsenyjin for provoking the discovery of yet another bug during the hackathon.

**Comment by palango** [comment]
@arsenyjin I'm investigating this. In the logs I see that the token address you registered is [0xf5b9359CC8e83b24e0436e78E6006608B53B5F46](https://kovan.etherscan.io/address/0xf5b9359CC8e83b24e0436e78E6006608B53B5F46#code), which is a `MintableToken` and not the `GRT` contract. Any reason for this?

Just wondering because the backend crashes when it tries the (non-existant) infos, which shouldn't happen of course.

```
contract GRT is MintableToken {
  string public constant name = "Granny Token";
  string public constant symbol = "GRT";
  uint8 public constant decimals = 18;
```

**Comment by palango** [comment]
This is fixed with #139 , but I still need to understand why parity throws that error.

**Comment by palango** [comment]
Seems to run fine now.
